controllers/util/mw_util.go

```diff
@@ -489,20 +489,59 @@ func (mwu *MWUtil) createOrUpdateManifestWork(
 	return nil
 }
 
-func (mwu *MWUtil) DeletePVManifestWork(fromCluster string) error {
+func (mwu *MWUtil) DeleteManifestWorksForCluster(clusterName string) error {
+	err := mwu.deleteVRGManifestWork(clusterName)
+	if err != nil {
+		mwu.Log.Error(err, "failed to delete MW for VRG")
+
+		return fmt.Errorf("failed to delete ManifestWork for VRG in namespace %s (%w)", clusterName, err)
+	}
+
+	err = mwu.deletePVManifestWork(clusterName)
+	if err != nil {
+		mwu.Log.Error(err, "failed to delete MW for PVs")
+
+		return fmt.Errorf("failed to delete ManifestWork for PVs in namespace %s (%w)", clusterName, err)
+	}
+
+	err = mwu.deleteRolesManifestWork(clusterName)
```

Comment:
Note: We don't need to delete the roles MWs as @ShyamsundarR indicated.  The roles could be used by other apps.